### PR TITLE
CASSANDRA-9409: Add tests to ensure keyspace isolation for UDA, UDF and UDT

### PR DESCRIPTION
**Pull request by aboudreault**

### CODE REVIEW

**Comment by aboudreault** [comment]
@ptnapoleon or @mambocab to review.

File: user_functions_test.py [hunk 1]
```diff
@@ -194,3 +194,38 @@ def udf_with_udt_test(self):
         assert_one(session, "select sum(funk(udt)) from tab", [6])
 
         assert_invalid(session, "drop type test;")
+
+    @since('2.2')
+    def udf_with_udt_keyspace_isolation_test(self):
+        "CASSANDRA-9409 - Ensure function dont allow a UDT from another keyspace"
```
Comment (ptnapoleon) [inline user_functions_test.py]: These need to be docstrings, so use `""""`, not just `"`". Also, if you could format it according to the contribution guidelines, that would be great.

**Comment by ptnapoleon** [comment]
+1 to the tests, but you need to make the docstring fixes first.

**Comment by aboudreault** [comment]
Good point for the docstring formatting. Fixed.

File: user_functions_test.py [hunk 2]
```diff
@@ -194,3 +194,45 @@ def udf_with_udt_test(self):
...
         assert_invalid(session, "drop type test;")
+
+    @since('2.2')
+    def udf_with_udt_keyspace_isolation_test(self):
+        """
+        Ensure functions dont allow a UDT from another keyspace
+        @jira_ticket CASSANDRA-9409
+        @since 2.2
+        """
+        session = self.prepare()
```
Comment (aboudreault) [inline user_functions_test.py]: @ptnapoleon  is the docstring ok like that? Or I should really ad expected_result etc.?

**Comment by mambocab** [comment]
LGTM too.

File: user_functions_test.py (hunk 2 above)
Comment (ptnapoleon) [inline user_functions_test.py]: This is perfect

**Pull request merged**